File: 1/part1.jl
input = [parse(Int, ss) for ss in readlines("input.txt")]

size = length(input)

for i in 1:size
  for j in (i + 1):size
    if input[i] + input[j] == 2020
      println("$(input[i]) * $(input[j]) = $(input[i] * input[j])")
    end
  end
end

File: 9/part1.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

namespace Day9
{
    class Part1
    {
        public static void Main(string[] args)
        {
            var input = File.ReadAllLines(@"input.txt").Select(ulong.Parse).ToList();

            for (var i = 25; i < input.Count; i++)
            {
                var value = input[i];
                var list = input.Skip(i - 25).Take(25).ToList();
                var found = false;
                for (var j = 0; j < list.Count; j++)
                {
                    for (var k = j + 1; k < list.Count; k++)
                    {
                        if (list[j] + list[k] == value)
                        {
                            found = true;
                        }
                    }
                }
                if (!found)
                {
                    Console.WriteLine(value);
                    break;
                }
            }
        }
    }
}
